SpatialGDK/ExternalSchemaCodegen/Programs/Improbable.CodeGen.Unreal/HeaderGenerator.cs

```diff
@@ -0,0 +1,376 @@
+using Improbable.CodeGen.Base;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Improbable.CodeGen.Unreal
+{
+    public static class HeaderGenerator
+    {
+        private static string DefaultHeaderIncludes = string.Join(Environment.NewLine, new []{
+            "#include <cstddef>",
+            "#include <cstddef>",
+            "#include <cstdint>",
+            "#include <functional>",
+            "#include <set>",
+            "#include <string>",
+            "#include \"CoreMinimal.h\"",
+            "#include \"Utils/SchemaOption.h\"",
+            "#include <WorkerSDK/improbable/c_schema.h>",
+            "#include <WorkerSDK/improbable/c_worker.h>",
+            $"#include \"{UnrealGenerator.RelativeIncludePrefix}{HelperFunctions.HeaderPath}\""
+        });
+    
+        public static string GenerateHeader(TypeDescription type, List<TypeDescription> types, Dictionary<string, TypeGeneratedCode> allGeneratedTypeContent, Bundle bundle)
+        {
+            var builder = new StringBuilder();
```

Comment:
Minor nit, but maybe we could have the GenerateFunctions be passed this string builder, then we don't have to create a new one in each function